module-services/service-time/AlarmOperations.cpp

```diff
@@ -293,9 +295,24 @@ namespace alarms
     void AlarmOperationsCommon::processOngoingEvents()
     {
         if (!ongoingSingleEvents.empty()) {
-            switchAlarmExecution(*(ongoingSingleEvents.front()), true);
-            handleActiveAlarmsCountChange();
-            handleSnoozedAlarmsCountChange();
+            if (onCheckIfPhoneCallIsOngoingCallback && onCheckIfPhoneCallIsOngoingCallback()) {
```

Comment:
Try to make it with an early return pattern to improve readability
for example:
if(ongoingSingleEvents.empty())
{return;}
and so on.
It will make the code more flat and easier to understand what is going on:)

Reply:
Ok, You right.

A changed this:
```cpp
    void AlarmOperationsCommon::processOngoingEvents()
    {
        if (!ongoingSingleEvents.empty()) {
            if (onCheckIfPhoneCallIsOngoingCallback && onCheckIfPhoneCallIsOngoingCallback()) {
            if (!ongoingSingleEvents.front()->wasHandledDuringPhoneCall && onAlarmDuringPhoneCallCallback) {
                    onAlarmDuringPhoneCallCallback();
                    ongoingSingleEvents.front()->wasHandledDuringPhoneCall = true;
                }
                OnSnoozeRingingAlarm onSnoozeRingingAlarmCallback = [](bool success) {
                    success ? LOG_DEBUG("Alarm snoozed during phone call ") : LOG_ERROR("Unable to snooze alarm!");
                };
                snoozeRingingAlarm(ongoingSingleEvents.front()->parent->ID,
                                   std::chrono::floor<std::chrono::minutes>(TimePointNow()) + std::chrono::minutes(1),
                                   onSnoozeRingingAlarmCallback);
            }
            else {
                ongoingSingleEvents.front()->wasHandledDuringPhoneCall = false;
                switchAlarmExecution(*(ongoingSingleEvents.front()), true);
                handleActiveAlarmsCountChange();
                handleSnoozedAlarmsCountChange();
            }
        }
    }
```
    
to this:
```cpp
    void AlarmOperationsCommon::processOngoingEvents()
    {
        if (ongoingSingleEvents.empty()) {
            return;
        }

        if (onCheckIfPhoneCallIsOngoingCallback && onCheckIfPhoneCallIsOngoingCallback()) {

            // Handle once during a phone call
            if (!ongoingSingleEvents.front()->wasHandledDuringPhoneCall && onAlarmDuringPhoneCallCallback) {
                onAlarmDuringPhoneCallCallback();
                ongoingSingleEvents.front()->wasHandledDuringPhoneCall = true;
            }

            // Snooze Alarm during a phone call
            OnSnoozeRingingAlarm onSnoozeRingingAlarmCallback = [](bool success) {
                success ? LOG_DEBUG("Alarm snoozed during a phone call ") : LOG_ERROR("Unable to snooze alarm!");
            };
            snoozeRingingAlarm(ongoingSingleEvents.front()->parent->ID,
                               std::chrono::floor<std::chrono::minutes>(TimePointNow()) + std::chrono::minutes(1),
                               onSnoozeRingingAlarmCallback);
            return;
        }

        ongoingSingleEvents.front()->wasHandledDuringPhoneCall = false;
        switchAlarmExecution(*(ongoingSingleEvents.front()), true);
        handleActiveAlarmsCountChange();
        handleSnoozedAlarmsCountChange();
    }
```